### PR TITLE
fix(postgrest): Update parameter type of `match()` filter so that null values cannot be passed. 

### PR DESCRIPTION
**Pull request by dshukertjr**
## What kind of change does this PR introduce?

Fixes https://github.com/supabase/supabase-flutter/issues/917

## What is the current behavior?

Currently, users can pass `null` to `match()` filter. 

## What is the new behavior?

Users will not be able to pass `null` to `match()`.

### CODE REVIEW

**Review by bdlukaa — APPROVED** [review]

**Comment by Vinzent03** [comment]
But the type of a map may be easily `Map<dynamic,dynamic>` despite having no null entries. So a working code might break now. I would prefer to note it in the documentation and use an assert over the actual values instead.

**Comment by dshukertjr** [comment]
Fair enough.

File: packages/postgrest/lib/src/postgrest_filter_builder.dart [hunk 1]
```diff
@@ -478,8 +480,10 @@ class PostgrestFilterBuilder<T> extends PostgrestTransformBuilder<T> {
   /// });
   /// ```
   PostgrestFilterBuilder<T> match(Map query) {
+    assert(!query.values.contains(null),
+        '`null` equality does not work with match. Use isFilter instead.');
     var url = _url;
-    query.forEach((k, v) => url = appendSearchParams('$k', 'eq.$v', url));
+    query.forEach((k, v) => url = appendSearchParams(k, 'eq.$v', url));
```
Comment (Vinzent03) [inline packages/postgrest/lib/src/postgrest_filter_builder.dart:486]: Might be good to undo this.
```suggestion
    query.forEach((k, v) => url = appendSearchParams('$k', 'eq.$v', url));
```

**Review by Vinzent03 — CHANGES_REQUESTED** [review]

**Comment by bdlukaa** [comment]
I like the first proposed solution better. It gives the developers type-safety, causing less hassle. It might be good to have this shipped in the next major release.

**Comment by dshukertjr** [comment]
>I like the first proposed solution better. It gives the developers type-safety, causing less hassle.

I like it too. I think sometimes we try to be a bit too picky about what defines a breaking change. I feel like the world does revolve around these minor "technically breaking changes" being shipped as a bug fix all the time. Flutter itself ships "breaking changes" without bumping the major versions all the time, and as long as it's under moderation, I don't think we should hesitate to ship these changes. 

I'd say we just ship it with the original change.

**Review by Vinzent03 — APPROVED** [review]
I don't think you can compare Flutter breaking changes with package breaking changes. Packages should use semver, Flutter doesn't.
I'm pretty sure three are code basis, which throw a static analysis error now, which definitely is a breaking change. But sure, our other bug fixes may sometimes be more breaking, and they  don't even show up when compiling.
I wouldn't make this breaking change, but you decide.

**Pull request merged**